### DISCUSSION TITLE
Geometry optimizaton

### DISCUSSION THREAD

**Discussion by am-tc01** [idaholab/moose#25471] [Q&A Modules: General]
Hi,
I'm working on a problem where I need to optimize the geometry i.e. I need to generate a new mesh on every Forward call from the Optimize Multiapp based on the optimization parameters. If I have understood correctly, as of now, the mesh is generated only once, at the start of the problem, and can't be re-created. I tried to use the `reset_apps` option in `FullSolveMultiApp` but following a previous [discussion](https://github.com/idaholab/moose/discussions/23945), and also in my experience, that doesn't seem to work. Is there any other way to re-generate the mesh at every Forward call?
Thanks!

**Comment by lindsayad** [comment]
Do you want to make meshes with different numbers of elements and/or nodes? If not you could displace the nodes to reproduce the mesh you want

**Reply by am-tc01** [reply]
I actually want to regenerate the entire mesh that has different numbers of elements and nodes.

**Reply by hugary1995** [reply]
What's the "driver" of the optimization algorithm? Is it the moose optimization module or something else?

**Reply by am-tc01** [reply]
It is the moose optimization module.

**Reply by lynnmunday** [reply]
This won't do what you are wanting with a new mesh but you might find it useful, @maxnezdyur  has a PR for shape optimization:  https://github.com/idaholab/moose/pull/24934#event-10373797083
He added a shape optimization test to that commit: modules/optimization/test/tests/executioners/constrained/shape_optimization/forward.i
It moves the mesh around with a diffusion problem.
